spec/support/shared_contexts.rb

```diff
@@ -0,0 +1,9 @@
+RSpec.shared_context "test mailer" do
+  class TestMailer < ActionMailer::Base
```

Comment:
I don't like this idea actually, Rubocop is right - https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Lint/ConstantDefinitionInBlock

Defining classes with let also complicates things.

Classes should be defined within modules. Moving it out to a shared module.